src/util/amount.rs

```diff
@@ -153,7 +173,13 @@ impl fmt::Display for ParseAmountError {
             ParseAmountError::InvalidCharacter(c) => write!(f, "invalid character in input: {}", c),
             ParseAmountError::UnknownDenomination(ref d) => write!(f, "unknown denomination: {}", d),
             ParseAmountError::PossiblyConfusingDenomination(ref d) => {
-                write!(f, "the 'M' at the beginning of {} should technically mean 'Mega' but that denomination is uncommon and maybe 'milli' was intended", d)
+                let (upper, lower) = match d.chars().next() {
+                    Some('M') => ("Mega", "milli"),
+                    Some('P') => ("Peta", "pico"),
+                    // This panic could be avoided by adding enum ConfusingDenomination { Mega, Peta } but is it worth it?
+                    _ => panic!("invalid error information"),
+                };
+                write!(f, "the '{}' at the beginning of {} should technically mean '{}' but that denomination is uncommon and maybe '{}' was intended", d.chars().next().unwrap(), d, upper, lower)
```

Comment:
Damn, sorry for imperfect suggestion, this would've been nicer:

```rust
 let (letter, upper, lower) = match d.chars().next() {
    Some('M') => ('M', "Mega", "milli"),
    Some('P') => ('P', "Peta", "pico"),
    // This panic could be avoided by adding enum ConfusingDenomination { Mega, Peta } but is it worth it?
    _ => panic!("invalid error information"),
};
write!(f, "the '{}' at the beginning of {} should technically mean '{}' but that denomination is uncommon and maybe '{}' was intended", letter, d, upper, lower)
```

Reply:
good idea. I updated it. sorry that u have to ack again.